modules/standard/Socket.md

```diff
@@ -0,0 +1,210 @@
+port: Integer - port number
+family: Integer - AF_INET or AF_INET6 or AF_UNSPEC
+```
+
+The purpose of family is important for address resolution and converting addr string to standard notation as well as validation. How the family can affect address resolution depends on [getaddrinfo](https://www.ibm.com/docs/en/zos/2.1.0?topic=functions-getaddrinfo-get-address-information#getaddrinfo) function.
+
+The `tcpConn` instance is basically a chapel `file` instance with added methods aligned to its nature as a socket. All the usual `file` methods are available on it:
+- `reader`
+  will return a reader channel for associated socket
+  ```python
+  proc tcpConn.writer() : channel
+  ```
+- `writer`
+  will return a writer channel for associated socket
+  ```python
+  proc tcpConn.writer() : channel
+  ```
+- `close` calling this on `tcpConn` will close the socket connection and associated `file` too.
+  ```python
+  proc tcpConn.close()
+  ```
+- `addr` this function will return a `ipAddr` object which will contain the `host` and `port` where socket is bound.
+
+## Creating a Listener
+
+`listen` function will create a `tcpListener` Instance which will take in following parameters
+
+**Method:**
+```python
+proc listen(addr:ipAddr, backlog=5, reuseaddr=true) : tcpListener
+```
+
+**Parameters**
+```
+addr: ipAddr - contains family info of socket to create and where to bind it.
+backlog: int - optional parameter, default value = 0
+reuseaddr: Boolean - optional parameter, default value = true
+```
+
+- The `backlog` argument defines the maximum length to which specifies the queue length for completely established sockets waiting to be accepted. If a connection request arrives when the queue is full, the client may receive an error with an indication of `ECONNREFUSED` or, if the underlying protocol supports retransmission, the request may be ignored so that a later reattempt at connection succeeds.
+- Calling in `listen` without `reuseaddr` can return an error that some other socket is already bound to the specified address. This can also be a result of re-running the program after only a short waiting period where the `bind` may find address as already in use.
+
+The `tcpListener` instance will have methods on it to call
+- `accept`: method will return a `tcpConn` instance of any new incoming connection otherwise will yield if no new connection is available at the instant. The communication can then proceed through the `tcpConn` instance returned
+  ```python
+  proc tcpListener.accept() : tcpConn
+  ```
+- `close`,
+  calling close on `tcpListener` will close the socket server and it won't accept anymore new connections.
+  ```python
+  proc tcpListener.close()
+  ```
+- `family` returns the family of socket `AF_INET` or `AF_INET6`
+- `addr` will return an `ipAddr` instance providing details about the address and port at which socket is listening.
+  ```python
+  proc tcpListener.addr() : ipAddr
```

Comment:
This could meet the cut for our current thinking about when should we omit parentheses in methods: https://github.com/chapel-lang/chapel/blob/main/doc/rst/developer/bestPractices/StandardModuleStyle.rst#functions-and-methods

Reply:
Thanks for pointing this out 😄  I actually made the construct in actual [code](https://github.com/chapel-lang/chapel/pull/17960/files#diff-25e31a95ef0ebf2378f1bf89a17c4e717d7ea626f91d109b43530a44f25ebe31R156) just like you suggested just need to update this doc